File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java
class BirdWatcher {
    private final int[] birdsPerDay;
    private int today;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
        today = this.birdsPerDay.length - 1;
    }

    public int[] getLastWeek() {
        return new int[] { 0, 2, 5, 3, 7, 8, 4 };
    }

    public int getToday() {
        return birdsPerDay[today];
    }

    public void incrementTodaysCount() {
        birdsPerDay[today]++;
    }

    public boolean hasDayWithoutBirds() {
        for (int day: birdsPerDay) {
            if (day == 0) {
                return true;
            }
        }

        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int total = 0;

        if (numberOfDays > birdsPerDay.length) {
            numberOfDays = birdsPerDay.length;
        }

        for (int i = 0; i < numberOfDays; i++) {
            total += birdsPerDay[i];
        }

        return total;
    }

    public int getBusyDays() {
        int busyDays = 0;

        for (int day: birdsPerDay) {
            if (day >= 5) {
                busyDays++;
            }
        }

        return busyDays;
    }
}

File: solutions/csharp/bird-watcher/2/BirdWatcher.cs
using System.Linq;

public class BirdCount
{
    private readonly int[] _birdsPerDay;

    public BirdCount(int[] birdsPerDay)
        => _birdsPerDay = birdsPerDay;

    public static int[] LastWeek()
        => new int[] { 0, 2, 5, 3, 7, 8, 4 };

    public int Today()
        => _birdsPerDay.Last();

    public void IncrementTodaysCount()
        => _birdsPerDay[_birdsPerDay.Length - 1]++;

    public bool HasDayWithoutBirds()
        => _birdsPerDay.Any(x => x == 0);

    public int CountForFirstDays(int numberOfDays)
        => _birdsPerDay.Take(numberOfDays).Sum();

    public int BusyDays()
        => _birdsPerDay.Count(x => x >= 5);
}
